src/trsp/pgr_trspHandler.cpp

```diff
@@ -94,10 +94,11 @@ void Pgr_trspHandler::clear() {
 // -------------------------------------------------------------------------
 double Pgr_trspHandler::construct_path(int64_t ed_id, Position pos) {
     pgassert(pos != ILLEGAL);
+    if (pos == ILLEGAL) return (std::numeric_limits<double>::max)();
```

Comment:
just curious why you added this because this condition is always false ... that said, I seem to remember that we would do this when working on `pickDeliver` optimization ...

Reply:
Because trsp is code I did not write and `Position` type has negative value and when you do access a vector v[-1] it will do a segmentation fault. (the pgassert compiling with `Debug` with throw and not create a segmentation fault`
you do not need to do that kind of if when the index to a vector is of size_t, because that type does not allow you to have negative numbers.